### DISCUSSION TITLE
w! triggers SaveAs, how can it be mapped to Save?

### DISCUSSION THREAD

**Discussion by bytebeast** [vscode-neovim/vscode-neovim#1066] [Q&A] [answered]
The `w!` pops up `SaveAs` window and populates filename. I have to move my hands away from keyboard to the mouse and click on the `Save` button.  Which breaks my flow of things. How can I change to behavior to trigger `Save` instead of `SaveAs` ?

**Comment by bytebeast** [comment]
i tried playing around with `vscode-file-commands.vim` this weekend, hehehe , but dunno enough about how to construct statement

**Comment by theol0403** [comment]
- why can't you just use `:w`?
- why can't you press enter to click the `Save` button in the `SaveAs` window?

**Reply by bytebeast** [reply]
`why can't you just use :w?`
> Oh, I have been trying to change my habit/muscle-memory.  It just takes some time to break a habit built over 20+ years

`why can't you press enter to click the Save button in the SaveAs window?`
> Pressing the `Enter/Return` key does appear to auto-press the `Save` key, but then another window appears with:
> "FILENAME", already exists. Do you want to replace it? Then with an option to `Cancel` or `Replace`. Pressing `Enter/Return` here auto-presses `Cancel`, and you return to the previous `Save` window.

**Comment by theol0403** [comment, marked as answer]
Use `command! -complete=file -bang -nargs=? Write call VSCodeNotify('workbench.action.files.save')` in your init.vim.

**Reply by bytebeast** [reply]
This works perfectly! Thanks!
